### PR TITLE
AWS S3 plugin doc

### PR DESCRIPTION
**Pull request by sumitsum**
1. Documentation for AWS S3 plugin.
2. Order database reference in Summary page alphabetically.

### CODE REVIEW

**Comment by trishaanand** [comment]
@sumitsum : In the example dynamic bindings, could we add examples of FilePicker for uploading a file?

**Comment by sumitsum** [comment]
> @sumitsum : In the example dynamic bindings, could we add examples of FilePicker for uploading a file?

Done.

File: datasource-reference/querying-amazons3.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Querying AWS S3
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply datasource-reference/querying-amazons3.md]: Done.

File: datasource-reference/querying-amazons3.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Appsmith supports the following actions on any AWS S3 instance:
+1. List all files in a bucket.
+2. Create a new file.
+3. Upload file.   
+4. Read file.
+5. Delete file.
+
+### List all files in a bucket
+1. set the 'Action' field to 'List all files in a bucket'
```
[the remark this answers is not in the document]
Comment (sumitsum) [reply datasource-reference/querying-amazons3.md]: Done.

**Comment by sumitsum** [comment]
Added another section describing how to download file.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**